samples/boot/saml2login/src/main/java/sample/SecurityConfig.java

```diff
@@ -33,4 +43,22 @@ RelyingPartyRegistrationRepository relyingPartyRegistrationRepository() {
 				.build();
 		return new InMemoryRelyingPartyRegistrationRepository(relyingPartyRegistration);
 	}
+
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+		http.authorizeRequests(request -> request.anyRequest().authenticated())
+				.saml2Login(Customizer.withDefaults()).apply(
+				new SecurityConfigurerAdapter<DefaultSecurityFilterChain, HttpSecurity>() {
```

Comment:
Let's simplify this to:

```java
@Bean 
Saml2MetadataFilter metadataFilter(RelyingPartyRegistrationRepository registrations) {
    DefaultRelyingPartyRegistrationResolver resolver = new DefaultRelyingPartyRegistrationResolver(registrations);
    return new Saml2MetadataFilter(resolver, new OpenSamlMetadataResolver());
}
```

I think the filter chain configuration can be removed at that point.

Also, in the sample, let's leave out the custom filename. The reason is that we want the samples to be a very simple introduction to Spring Security.